core/src/test/scala/org/apache/spark/sql/delta/TimestampLocalFileSystem.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import java.net.URI
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{DelegateToFileSystem, Path, RawLocalFileSystem}
+import org.apache.hadoop.fs.FileStatus
+
+/**
+ * This custom fs implementation is used for testing the msync calling in HDFSLogStore writes.
+ * If `msync` is not called, `listStatus` will return stale results.
+ */
+class TimestampLocalFileSystem extends RawLocalFileSystem {
+
+  private var uri: URI = _
+  private var latestTimestamp: Long = 0
+
+  override def getScheme: String = TimestampLocalFileSystem.scheme
+
+  override def initialize(name: URI, conf: Configuration): Unit = {
+    uri = URI.create(name.getScheme + ":///")
+    latestTimestamp = System.currentTimeMillis()
```

Comment:
If we remove this, do we still need `Thread.sleep(1000)` in the test (in other words, initialize `latestTimestamp` to 0)? My concern is `Thread.sleep(1000)` is not guaranteed to sleep 1000ms, which may make this test flaky.

Reply:
I added this to be slightly more representative of a file system msync'ing on startup, but if I just set this to 0 then I think the existing test _would_ still pass without the sleep so I can just do that